src/Network/Ethereum/Web3/Provider.hs

```diff
@@ -14,30 +14,24 @@ module Network.Ethereum.Web3.Provider where
 import           Control.Concurrent.Async    (Async, async)
 import           Control.Exception           (try)
 import           Control.Monad.IO.Class      (MonadIO (..))
+import           Control.Monad.Trans.Reader  (mapReaderT, runReaderT)
 import           Network.Ethereum.Web3.Types
-
--- | Ethereum node service provider
-class Provider a where
-    -- | JSON-RPC provider URI, default: localhost:8545
-    rpcUri :: Web3 a String
-
--- | Default 'Web3' service provider
-data DefaultProvider
-
-instance Provider DefaultProvider where
-    rpcUri = return "http://localhost:8545"
+import           Network.HTTP.Client         (newManager)
+import           Network.HTTP.Client.TLS     (tlsManagerSettings)
 
 -- | 'Web3' monad runner
-runWeb3' :: MonadIO m => Web3 a b -> m (Either Web3Error b)
+runWeb3' :: MonadIO m => ServerUri -> Web3 a -> m (Either Web3Error a)
```

Comment:
Just define `Web3Config` datatype with `Default` instance.

Reply:
for the Web3Config, are you looking for something like 
```haskell
newtype Web3Config = Web3Config { unWeb3Config :: ServerUri }
```

Reply:
Yes, to improve performance we can use `newtype` here.

But, in my opinion, more generic way is define something like

```haskell
data Provider = HttpProvider ServerUri | IpcProvider FilePath
  deriving (Show, Eq, Generic)

data Web3Config = Web3Config {
  provider :: Provider
} deriving (Show, Eq, Generic)

Reply:
Okay, seems straightforward.  Seems like we could scrap `Web3Config` and just use `Provider` defining the Default instance on `Provider`

Reply:
Also, my current implementation doesn't support Ipc. The example you posted, https://kseo.github.io/posts/2017-01-23-custom-connection-manager-for-http-client.html  requires a URI and a `FilePath` to the socket. I see `FilePath` is used to create the `Manager` but I'm not 100% sure what the `ServerUri` would be. For now I would like to define the Provider type as
```haskell
-- TODO: Add to type `IpcProvider FilePath` to support IPC
data Provider = HttpProvider ServerUri 
  deriving (Show, Eq, Generic)
```
 and add  support for IPC to the roadmap. How does this sound?

Reply:
Cool!

Also, may it'll be interested to add ability for users to pass custom `Manager`, for example:

```haskell
data Provider = HttpProvider ServerUri | CustomProvider Manager
  deriving (Show, Eq, Generic)
```